### PR TITLE
feat(staking,ui): Preferences modal, bug fixes

### PR DESCRIPTION
**Pull request by szymonmaslowski**
# Checklist

- [x] JIRA
   - https://input-output.atlassian.net/browse/LW-7544
   - https://input-output.atlassian.net/browse/LW-7638
   - https://input-output.atlassian.net/browse/LW-7660
   - https://input-output.atlassian.net/browse/LW-7661
   - https://input-output.atlassian.net/browse/LW-7662
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

# Staking
Scope described by linked tasks above.

# UI Toolkit
Improved the Subnavigation component by enabling it to support controlled mode.

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/1158/5674099506/index.html) for [925a24bf](https://github.com/input-output-hk/lace/pull/305/commits/925a24bf40a99940ad272bd225a951692ef59e7d)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 35     | 0      | 0       | 0     | 35    | ✅     |

### CODE REVIEW

**Review by renanvalentin — APPROVED** [review]
The changes on the UI package looks good!

File: packages/staking/src/features/drawer/StakePoolDetails.tsx [hunk 1]
```diff
@@ -50,20 +49,22 @@ export const StakePoolDetails = ({
...
     }),
-    [onStake, canDelegate]
+    [onSelect, onStakeOnThisPool, onUnselect]
   );
 
-  const pendingDelegationAndAnyDetailsAreShown =
-    isTherePendingDelegation && simpleSendConfig.currentSection === Sections.DETAIL;
   const cannotAddAnotherPoolToDraft = draftFull && !openPoolSelectedInDraft;
-  const footerHidden = cannotAddAnotherPoolToDraft || pendingDelegationAndAnyDetailsAreShown;
+  const footerHidden =
```
Comment (danielmain) [inline packages/staking/src/features/drawer/StakePoolDetails.tsx:65]: I would create a function called, shouldFooterBeDisplayed adding comments what conditions are checked. This "if" is very hard to understand. If no time for this, just a TODO with a Jira in our backlog with some comments

**Review by danielmain — CHANGES_REQUESTED** [review]

File: packages/staking/src/features/modals/StakingModal.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  return (
+    <Modal
+      destroyOnClose
+      centered
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      className={cn(styles.modal, { [styles.popupView!]: popupView })}
+      onCancel={handleCancelModal}
+      // eslint-disable-next-line unicorn/no-null
+      footer={null}
```
[the remark this answers is not in the document]
Comment (szymonmaslowski) [reply packages/staking/src/features/modals/StakingModal.tsx:50]: This way we disable the footer
image

File: packages/staking/src/features/staking/StakingView.tsx [hunk 3]
```diff
@@ -1,41 +1,97 @@
...
+      weight: 1,
+    });
+  }, [openPool, openPoolDetails, portfolioMutators, walletStoreWalletUICardanoCoin]);
+
+  const stakeJustOnCurrentPool = useCallback(() => {
+    if (alreadyDelegating) {
+      setPendingSelection(true);
+    }
+    initiateStaking();
+    // TODO: implement no funds modal
```
[the remark this answers is not in the document]
Comment (szymonmaslowski) [reply packages/staking/src/features/staking/StakingView.tsx]: Done 👍

**Review by danielmain — APPROVED** [review]

**Comment by ManusMcCole** [comment]
Hi @szymonmaslowski. Good job on rectifying all the issues on the cards. However this flow doesnt have any Next button so the user can succesfully complete the process to delegate to multiple pools. See [video](https://drive.google.com/file/d/18hcOha9fKHoUf2A1P4zNI-wC_8Sa5tYH/view?usp=sharing)

**Comment by szymonmaslowski** [comment]
> Hi Szymon. I tested newest build and the Next button is visible and clickable but results in a state of constant loading that never resolves. This happens for all delegation. See [Video](https://drive.google.com/file/d/1LkUhoUf3S5VU0O6RKeiJGhTHmeZuqUxr/view?usp=sharing)

Hey Manus. I cannot view the video because of lack of permissions. However I assume you are talking about the next button on the confirmation drawer (the one with the list of selected pool and fee) being blocked and having the spinner visible. That is a known issue and it is present also in the old staking. Please try to switch the network to overcome that issue.

**Review by ManusMcCole — APPROVED** [review]

**Pull request merged**